File: frontend/src/api/export.ts
import request from './axios'
import { unwrapApiData, type ApiResponse } from './response'

// ==================== 类型定义 ====================

export type ExportFormat = 'csv' | 'json' | 'xlsx'
export type ExportType = 'task_logs' | 'cloud_stats' | 'exchange_records' | 'accounts' | 'all'

export interface ExportHistoryItem {
  id: number
  user_id: number
  type: ExportType
  format: ExportFormat
  status: string
  file_path?: string
  file_size?: number
  created_at: string
  completed_at?: string
}

export interface ExportHistoryResponse {
  exports: ExportHistoryItem[]
  total: number
}

// ==================== API 函数 ====================

// 导出数据
export function exportData(params: {
  type: ExportType
  format?: ExportFormat
  account_id?: number
  start_date?: string
  end_date?: string
  status?: string
}): Promise<Blob> {
  return request<Blob>({
    url: '/api/export',
    method: 'get',
    params,
    responseType: 'blob'
  })
}

// 获取导出历史
export function getExportHistory(): Promise<ExportHistoryResponse> {
  const fallback: ExportHistoryResponse = { exports: [], total: 0 }
  return request<ExportHistoryResponse | ApiResponse<ExportHistoryResponse>>({
    url: '/api/export/history',
    method: 'get'
  }).then((res) => unwrapApiData(res, fallback))
}

File: backend/internal/domain/entity/export.go
package domain

import "time"

// ExportFormat represents supported export formats
type ExportFormat string

const (
	ExportFormatCSV  ExportFormat = "csv"
	ExportFormatJSON ExportFormat = "json"
	ExportFormatXLSX ExportFormat = "xlsx"
)

// ExportType represents what data to export
type ExportType string

const (
	ExportTypeTaskLogs      ExportType = "task_logs"
	ExportTypeCloudStats    ExportType = "cloud_stats"
	ExportTypeExchangeRecords ExportType = "exchange_records"
	ExportTypeAccounts      ExportType = "accounts"
	ExportTypeAll           ExportType = "all"
)

// ExportJob represents an export job request
type ExportJob struct {
	ID          uint
	UserID      uint
	Type        ExportType
	Format      ExportFormat
	Filters     map[string]interface{}
	Status      string
	FilePath    string
	FileSize    int64
	ExpiresAt   time.Time
	CreatedAt   time.Time
	CompletedAt *time.Time
}

// WebhookEventType represents webhook event types
type WebhookEventType string

const (
	WebhookEventTaskSuccess  WebhookEventType = "task.success"
	WebhookEventTaskFailure  WebhookEventType = "task.failure"
	WebhookEventExchangeHit  WebhookEventType = "exchange.hit"
	WebhookEventSystemAlert  WebhookEventType = "system.alert"
)

// WebhookEndpoint represents a webhook endpoint
type WebhookEndpoint struct {
	ID          uint
	UserID      uint
	Name        string
	URL         string
	Events      []WebhookEventType
	Secret      string
	IsActive    bool
	Headers     map[string]string
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// PlatformType represents supported cloud platforms
type PlatformType string

const (
	PlatformTypeCloud189   PlatformType = "cloud_189"
	PlatformTypeCloud115   PlatformType = "cloud_115"
	PlatformTypeAliyun     PlatformType = "aliyun"
	PlatformTypeQuark      PlatformType = "quark"
	PlatformTypeBaidu      PlatformType = "baidu"
)

// PlatformAccount represents a cloud platform account
type PlatformAccount struct {
	ID          uint
	UserID      uint
	Platform    PlatformType
	Username    string
	DisplayName string
	AuthData    map[string]interface{} // encrypted auth tokens
	IsActive    bool
	LastSyncAt  *time.Time
	CreatedAt   time.Time
	UpdatedAt   time.Time
}
